custom/src/main/java/org/kiwix/kiwixmobile/custom/download/CustomDownloadFragment.kt

```diff
@@ -112,15 +116,40 @@ class CustomDownloadFragment : BaseFragment(), FragmentActivityExtensions {
 
       is DownloadFailed -> {
         fragmentCustomDownloadBinding?.cdViewAnimator?.setDistinctDisplayedChild(2)
-        fragmentCustomDownloadBinding?.customDownloadError?.cdErrorText?.text =
-          context?.let(state.downloadState::toReadableState)
+        val errorMessage = context?.let { context ->
+          if (state.downloadState.zimUrl?.isAuthenticationUrl == false) {
+            return@let getErrorMessageFromDownloadState(state.downloadState, context)
+          }
+
+          val defaultErrorMessage = getErrorMessageFromDownloadState(state.downloadState, context)
+          // Check if `REQUEST_NOT_SUCCESSFUL` indicates an unsuccessful response from the server.
+          // If the server does not respond to the URL, we will display a custom message to the user.
+          if (defaultErrorMessage == context.getString(
+              R.string.failed_state,
+              "REQUEST_NOT_SUCCESSFUL"
+            )
+          ) {
+            context.getString(
+              R.string.failed_state,
+              context.getString(R.string.custom_download_error_message_for_authentication_failed)
+            )
+          } else {
+            defaultErrorMessage
```

Comment:
@MohitMaliDeveloper I think it always gets this message from the fetch that we're talking about.

```
else {
            defaultErrorMessage
```
This code will never execute.

Reply:
> This code will never execute.

Hi @gouri-panda this code executes if there is another error that happens in fetch e.g. network error, CONNECTION_TIMED_OUT, UNKNOWN_HOST,  WRITE_PERMISSION_DENIED etc.